jep/0000/README.adoc

```diff
@@ -0,0 +1,264 @@
+
+```yaml
+metadata:
+  # labels and annotations are key: value string attributes
+  labels:
+    name: myplugin
+    groupId: ...
+    artifactId: ...
+    version: ...
+  annotations:
+    io.jenkins.x: y
+spec:
+  core:
+    # version OR version + ref (version just to keep Maven happy about version order)
+    ref: master
+    version: 1.0
+  components:
+    - groupId: org.acme
+      artifactId: acme-component
+      ref: master
+      version: 1.0
+  plugins:
+    - groupId: org.acme
+      artifactId: acme-plugin
+      ref: master
+      # version: 1.0
+  environments:
+    # environments get added to the other section when enabled
+    - name: aws
+      plugins:
+        - groupId: org.acme
+          artifactId: acme2-plugin
+          ref: master
+          # version: 1.0
+      components: ...
+  # other sections can be added and ignored by default
+# the realized BoM after refs are resolved
+status:
+  core:
+    ref: aaabbb
+    # version: 1.0
+  plugins:
+    - groupId: org.acme
+      artifactId: acme-plugin
+      ref: bbbccc
+      # version: 1.0
+  environments:
+    - name: aws
+      plugins:
+        - groupId: org.acme
+          artifactId: acme2-plugin
+          ref: cccddd
+          # version: 1.0
+```
+
+
+== Motivation
+
+The current approach to make changes in core, libraries and plugins is too cumbersome, far from Continuous Integration and complicated for contributors, due to the usage of multiple repositories.
+
+Changes typically span more than one repository, causing contributors to manually combine different PRs together.
+The goal of this proposal is to move towards a master based delivery process, ensure that core changes don't break plugins and that core changes needed by plugins can be quickly and safely adopted.
+
+This proposal builds on the goals of Essentials. We want to ensure that the Essentials distribution is continuously delivered, off master, and is done safely with a set of checks that run automatically.
+
+== Reasoning
+
+The chosen YAML format is just picked due to the similarities with Kubernetes model objects and has no importance.
+
+For version naming there are other options:
+
+* Use Maven SNAPSHOTS
+** Automatically deploy snapshots using commit ids (ie. jenkins-core:aabbcc-SNAPSHOT)
+** Ensure the commit ids are included in the packaging and visible during builds
+* Use git modules to point to the master and PR commits
+** And build everything every time
+** This would not work for Essentials as the components need to be individually downloadable.
+
+
+
+== Backwards Compatibility
+
+This proposal aims to add new functionality and reuse existing tooling by generating Maven poms and other formats in use today.
+
+== Security
+
+There are no security risks related to this proposal.
+
+== Infrastructure Requirements
+
+There are no new infrastructure requirements related to this proposal.
+
+== Testing
+
+There are no testing issues related to this proposal.
+
+== References
```

Comment:
https://github.com/jenkinsci/jep/tree/master/jep/1#required-sections . "Prototype Implementation" section is missing. Without that JEP cannot be accepted as draft. My proposal would be to create a prototype implementation in https://github.com/jenkinsci/custom-war-packager.